control-plane/subcommand/sync-catalog/command.go

```diff
@@ -68,6 +73,12 @@ type Command struct {
 	flagK8SNSMirroringPrefix       string   // Prefix added to Consul namespaces created when mirroring
 	flagCrossNamespaceACLPolicy    string   // The name of the ACL policy to add to every created namespace if ACLs are enabled
 
+	// Metrics settings.
+	flagEnableMetrics        string
```

Comment:
if we make this a bool instead of a string then we don't need to do any of the `isset` type stuff on parsing the value

Reply:
one thing to keep in mind is that when we call it with `false` (which we probably don't need to here because just setting `-enable-metrics` will set the value to true and it will default to false) buuuut to call it with an explicit false we'd need to do `-enable-metrics=false`
https://pkg.go.dev/flag#hdr-Command_line_flag_syntax